generate_instructions.py

```diff
@@ -187,19 +228,56 @@ def solve_board(current_config, goal_config):
     except NoActionException as e:
         return []
 
+def create_header_list(num_blocks):
+    header_ls = []
+    for i in range(num_blocks):
+        header_ls.append('block{0}_id'.format(i))
+        header_ls.append('block{0}_side1_letter'.format(i))
+        header_ls.append('block{0}_side1_color'.format(i))
+        header_ls.append('block{0}_side2_letter'.format(i))
+        header_ls.append('block{0}_side2_color'.format(i))
+        header_ls.append('block{0}_x_pos'.format(i))
+        header_ls.append('block{0}_y_pos'.format(i))
+    header_ls.append('moved_block_id')
+    header_ls.append('moved_block_x_pos')
+    header_ls.append('moved_block_y_pos')
+    header_ls.append('phrase')
+    header_ls.append('point_x')
+    header_ls.append('point_y')
+    return header_ls
+
 def main():
     colors = ['RED', 'GREEN', 'BLUE']
     letters = ['A', 'B', 'C']
-    start = random_configuration(2, letters, colors)
-    end = start.scatter().mark_complete()
+    num_blocks = 2
 
-    b = random_block(colors, letters)
-    b_prime = b.shift_to(random_position())
+    if len(sys.argv) != 3:
+        print('Wrong number of arguments.')
+        print('Call as "python generate_instructions.py <num-to-generate> <outfile>"')
+        return
 
-    print('start: %s' % start)
-    print('end: %s' % end)
-    print()
-    print('actions: %s' % str(list(map(str, solve_board(start, end)))))
+    try:
+        num_to_generate = int(sys.argv[1])
+    except:
+        print('First argument must be an integer.')
+        return
+
+    try:
+        f = open(sys.argv[2], 'w')
+        csv_writer = csv.writer(f)
+    except:
+        print('Given path is not valid or could not be opened')
+        return
+
+    csv_writer.writerow(create_header_list(num_blocks))
+    for _ in range(num_to_generate):
+        start = random_configuration(num_blocks, letters, colors)
+        end = start.scatter().mark_complete()
+        moves = solve_board(start, end)
+        for line_repr in map(lambda x: x.list_representation(), moves):
+            csv_writer.writerow(line_repr)
```

Comment:
I feel bad asking for this, but a sanity test that would check that we wrote out a `.csv` file in the correct format would be nice. If you've run it and it spat out something reasonable, though, we can probably accept that and move on.

Reply:
I can, although it feels to me like were just unit testing the library which doesn't make a whole lot of sense to me. Sorry for taking so long to see this. I can add that tonight or tomorrow

Reply:
Come to think of it, this code will never run automated in tandem with anything else; we'll run it, glance at the output, and stuff it into the NN. Any flaws should be really apparent at that point, so a full integration test really isn't going to help much.

I say forget about it.

Reply:
I'm not going to argue with that